docs/RecoveringFromDeepForks.md

```diff
@@ -0,0 +1,99 @@
+# Recovering from deep forks
+
+Help! My node has failed with the following fatal error:
+
+```
+Fatal error: Requested rewind exceeds limit (4). Our previous cut block height: 1, fork ancestor's block height: 6.
+Offending new block:
+BlockHeader {_blockNonce = Nonce 0, _blockCreationTime = BlockCreationTime {_bct = Time (TimeSpan (Micros 1575672858613932))}, _blockParent = "ugAOff6YkKmvp_pJIHqx6HLTRPwyHEHD2c2Olm6L2Fg", _blockAdjacentHashes = BlockHashRecord {_getBlockHashRecord = fromList []}, _blockTarget = HashTarget (PowHashNat 115792089237316195423570985008687907853269984665640564039457584007913129639935), _blockPayloadHash = BlockPayloadHash (MerkleLogHash U-3paUKNKQUkggFmNOnGBEAJnjpy4K_-7t_dGyV0lhw), _blockChainId = ChainId 8, _blockWeight = BlockWeight (HashDifficulty (PowHashNat 1)), _blockHeight = 1, _blockChainwebVersion = fastTimedCPM-peterson, _blockEpochStart = EpochStartTime (Time (TimeSpan (Micros 0))), _blockFlags = FeatureFlags 0, _blockHash = "SKZ0mTWp08ooXsWz_yyT7v_MSrRdyKkmCyWJBTK8u3g"}
+
+Your node is part of a losing fork longer than your reorg-limit, which
```

Comment:
This statement is not completely correct. For efficiency/DOS reasons we don't validate the forks all the way to the top. So we don't know which fork is actually winning. Doing so would be costly and could be exploited. We thus ask the node administrator to resolve which fork should be given preference.

The [chainweb-forks](https://github.com/kadena-community/chainweb-forks) tool is meant to help administrators to make this decision.